src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,196 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (head == null && tail == null && size == 0) {
+            addToEmptyList(value);
+        } else if (head != null && tail == null) {
+            addInside(value);
+        } else {
+            addIfLast(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index is bigger than size");
+        }
```

Comment:
and dont forget to specify index in error message
```suggestion
        checkIndexForAdd(index);
```